packages/volto/src/components/manage/Blocks/Search/hocs/withSearch.jsx

```diff
@@ -288,8 +325,17 @@ const withSearch = (options) => (WrappedComponent) => {
     );
     const previousUrlQuery = usePrevious(urlQuery);
 
+    // During first render the previousUrlQuery is undefined and urlQuery
+    // is empty so it ressetting the facet when you are navigating but during reload we have urlQuery and we need
+    // to set the facet at first render.
+    const preventOverrideOfFacetState =
+      previousUrlQuery === undefined && urlQuery.length === 0;
```

Comment:
I think trying to make decision here based on some previous state is not a good way to do things. In general. It makes the working of the block harder to reason about (because you introduce time-traveling by looking at the previous state), and harder to replicate, etc.

Now, I am guily of doing this as well, in other cases. But I always feel that it's not a good path to go.